src/FacturX.php

```diff
@@ -6,11 +6,82 @@
 
 class FacturX
 {
-//    private const VERSION = '1.7';
-//    private const ENCODING = 'UTF-8';
+    private string $pdfContent;
 
-    public function generate(): void
+    private ?string $xmlContent;
+
+    private \Atgp\FacturX\Facturx $facturxManager;
+
+    public function __construct(string $pdfContent, ?string $xmlContent = null)
     {
-        return;
+        $this->facturxManager = new \Atgp\FacturX\Facturx();
+
+        $pdfContainsValidXml = true;
+        try {
+            $this->facturxManager->getFacturxXmlFromPdf($pdfContent);
+        } catch (\Exception $e) {
+            $pdfContainsValidXml = false;
+        }
+
+        if ($xmlContent !== null) {
+            $isValidXml = true;
+
+            try {
+                $this->facturxManager->checkFacturxXsd($xmlContent);
+            } catch (\Exception $e) {
+                $isValidXml = false;
+            }
+        } else {
+            $isValidXml = false;
+        }
+
+        if (!$pdfContainsValidXml || !$isValidXml) {
+            throw new \Exception(
+                'FacturX must contain a PDF with a valid XML or be created with a valid XML to be inserted in the PDF.'
+            );
+        }
+
+        $this->pdfContent = $pdfContent;
+        $this->xmlContent = $xmlContent;
+    }
+
+    public function getXmlContent(): ?string
+    {
+        return $this->xmlContent;
+    }
+
+    public function getPdfContent(): string
+    {
+        return $this->pdfContent;
+    }
+
+    /**
+     * @param array<int, string> $additionalAttachments
+     */
+    public function generate(array $additionalAttachments = [], bool $addFacturxLogo = true): string
```

Comment:
```suggestion
    public function generate(array $additionalAttachments = [], bool $addFacturxLogo = true): string
```
redondant avec getPDFContent.

Pour qu'on comprenne mieux mes commentaires, je vois plus une API comme : 
```php
class FacturX {
  public function __construct(string $pdfContent, ?string $xmlContent = null);
  public function getXmlContent(): string;
  public function getPdfContent(): string;
  public function addFacturxLogo(): void;
  public function addAttachments(iterable $additionalAttachments):void;
}
```